contrib/natss/pkg/dispatcher/dispatcher.go

```diff
@@ -211,21 +213,34 @@ func (s *SubscriptionsSupervisor) UpdateSubscriptions(channel *eventingv1alpha1.
 		chMap = make(map[subscriptionReference]*stan.Subscription)
 		s.subscriptions[cRef] = chMap
 	}
+	var errStrings []string
 	for _, sub := range subscriptions {
+		subscriberStatus := ToSubscriberStatus(&sub, corev1.ConditionFalse, "internal error")
 		// check if the subscription already exist and do nothing in this case
 		subRef := newSubscriptionReference(sub)
 		if _, ok := chMap[subRef]; ok {
 			activeSubs[subRef] = true
 			s.logger.Sugar().Infof("Subscription: %v already active for channel: %v", sub, cRef)
+			subscriberStatus.Ready = corev1.ConditionTrue
+			subscriberStatus.Message = fmt.Sprintf("Subscription: %v already active for channel: %v", sub, cRef)
 			continue
 		}
 		// subscribe
 		natssSub, err := s.subscribe(cRef, subRef)
 		if err != nil {
-			return err
+			subscriberStatus.Ready = corev1.ConditionTrue
```

Comment:
here should be False?

Reply:
yes. last minute code refactor and copy paste error. WIll fix it. Thanks for catching.